### PR TITLE
13.0 test website visitor fja

### PR DESCRIPTION
**Pull request by fja-odoo**
Some website_visitor tests

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fja-odoo** [comment]
@JKE-be Rebased and fixed conflicts

File: odoo/tests/common.py [hunk 1]
```diff
@@ -1004,6 +1004,10 @@ def get_http_request_threads():
             self._logger.info('remaining requests')
             odoo.tools.misc.dumpstacks()
 
+    def logout(self, keep_db=True):
+        self.session.logout(keep_db=True)
+        odoo.http.root.session_store.save(self.session)
```
Comment (JKE-be) [inline odoo/tests/common.py]: can you refresh my memory about this last line ?

Comment (fja-odoo) [reply odoo/tests/common.py]: I need to logout the user to test if the pages visited after are tracked properly. I don't think there is another way to do so.

Comment (JKE-be) [reply odoo/tests/common.py]: nice to have a param keep_db unsued.
why not use it in logout() call to...

But btw, why can't you use the real logout.

If we patch 80% of the methods, we will only test 20% of the code ...

Comment (fja-odoo) [reply odoo/tests/common.py]: What do you mean by real Logout ? In an HTTPCase I don't have any current request so I have to do it manually.

**Comment by fja-odoo** [comment]
@JKE-be 20 builds passed the tests: https://runbot.odoo.com/runbot/build/2850760

**Comment by JKE-be** [comment]
@robodoo r+

**Pull request closed without merging**